src/Shared/Core/Validation/Check.cs

```diff
@@ -94,6 +94,21 @@ public static T NotNull<T>([NotNull] T? argument, [CallerArgumentExpression("arg
         return argument;
     }
 
+    /// <summary>
+    /// Checks if the provided string is an entity instance id, throwing if it is.
+    /// Throws <see cref="ArgumentException" /> if the conditions are not met.
+    /// </summary>
+    /// <param name="argument">The string to check.</param>
+    /// <param name="name">The name of the string for the exception.</param>
+    public static void NotEntity(
+        string? argument, [CallerArgumentExpression("argument")] string? name = default)
+    {
+        if (argument?.Length > 0 && argument[0] == '@')
+        {
+            throw new ArgumentException("Parameter cannot be an entity instance id", name);
```

Comment:
nit: should clarify what entity instance ID means:

`Orchestration instance IDs cannot start '@' when entity support is enabled. '@' is reserved for entities. Instance ID: '{argument}'.`